src/transformers/modeling_tf_utils.py

```diff
@@ -656,143 +769,177 @@ def resize_token_embeddings(self, new_num_tokens=None) -> tf.Variable:
         Return:
             :obj:`tf.Variable`: Pointer to the input tokens Embeddings Module of the model.
         """
-        model_embeds = self._resize_token_embeddings(new_num_tokens)
-        if new_num_tokens is None:
-            return model_embeds
+        if new_num_tokens is None or new_num_tokens == self.config.vocab_size:
+            return self._find_weights(self.get_input_embeddings())
 
-        return model_embeds
+        model_embeds = self._resize_token_embeddings(new_num_tokens)
 
-    def _resize_token_embeddings(self, new_num_tokens):
-        # get_input_embeddings and set_input_embeddings need to be implemented in base layer.
-        base_model = getattr(self, self.base_model_prefix, self)
-        old_embeddings = base_model.get_input_embeddings()
-        new_embeddings = self._get_resized_embeddings(old_embeddings, new_num_tokens)
-        base_model.set_input_embeddings(new_embeddings)
         # Update base model and current model config
         self.config.vocab_size = new_num_tokens
-        base_model.vocab_size = new_num_tokens
-        return base_model.get_input_embeddings()
-
-    def _get_word_embeddings(self, embeddings):
-        if hasattr(embeddings, "word_embeddings"):
-            # TFBertEmbeddings, TFAlbertEmbeddings, TFElectraEmbeddings
-            return embeddings.word_embeddings
-        elif hasattr(embeddings, "weight"):
-            # TFSharedEmbeddings
-            return embeddings.weight
+
+        return model_embeds
+
+    def _find_weights(self, embedding_layer):
+        if hasattr(embedding_layer, "word_embeddings"):
+            return embedding_layer.word_embeddings
+        elif hasattr(embedding_layer, "weight"):
+            return embedding_layer.weight
+        elif hasattr(embedding_layer, "decoder"):
+            return embedding_layer.decoder
```

Comment:
We should keep this method in mind when defining new LM head layers otherwise this would grow to be unmaintainable. It's unfortunate that we need to have such a mapping here, but I understand why it's necessary cf your comment [here](https://github.com/huggingface/transformers/pull/9193#discussion_r546039935).

Reply:
I'm currently rethinking the way we implement de embeddings and a part of this "rethinking" is how to get rid of these attribute detection, or at least minimize the side effects.

Reply:
Can we change the naming a bit maybe? just `_find_weights` is a bit too generic IMO => `_get_word_embedding_weight` ? Also (not sure here) maybe raise if nothing is found instead of silent None?